### PR TITLE
feat(Ida-98): Logging for IDA using AOP

### PR DESCRIPTION
**Pull request by maqboolkhan**
In this PR, Logging for IDA was introduced! AOP was used using Log4j2.

### CODE REVIEW

File: backend-server/src/main/java/org/dice/ida/model/ChatMessageResponse.java [hunk 1]
```diff
@@ -1,96 +1,110 @@
...
+		this.payload = payload;
+	}
+
+	public int getErrCode() {
+		return errCode;
+	}
+
+	public void setErrCode(int errCode) {
+		this.errCode = errCode;
+	}
```
Comment (nikit91) [inline backend-server/src/main/java/org/dice/ida/model/ChatMessageResponse.java:103]: You need to make sure that this field is written in the serialized json (ChatMsgRespSerializer.java)

File: backend-server/src/main/java/org/dice/ida/logging/AspectLogger.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	@Autowired
+	private ChatMessageResponse response;
+
+	/**
+	 * Creates the point cut for the MessageRestController
+	 *
+	 */
+	@Pointcut("execution(* org.dice.ida.controller.*.*(..))")
+	public void controller() {
+		//PointCut for MessageRestController
```
Comment (nikit91) [inline backend-server/src/main/java/org/dice/ida/logging/AspectLogger.java:34]: put a debug logging statement here.

File: backend-server/src/main/java/org/dice/ida/logging/AspectLogger.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		commonMsgBody(joinPoint, logMessage);
+		log.info(logMessage.toString());
+	}
+
+	/**
+	 * Method to log the exceptions
+	 * @param joinPoint
+	 * @param exception
+	 */
+	@AfterThrowing(pointcut = "execution(* org.dice.ida.*.*.*.*(..))", throwing = "exception")
```
Comment (nikit91) [inline backend-server/src/main/java/org/dice/ida/logging/AspectLogger.java:54]: This looks like a very specific pattern and is limited by the depth. Please use something like `"execution(* org.dice.ida..*(..))"` (Ref: https://www.baeldung.com/spring-aop-pointcut-tutorial#4-args)

**Review by nikit91 — CHANGES_REQUESTED** [review]
The PR mostly looks good. I have mentioned a few changes, please make sure those are taken care of.

**Comment by nandeeshp24** [comment]
There seems to be some issue building the application after these changes. @maqboolkhan Please build it at your end and check before this PR gets merged

**Comment by maqboolkhan** [comment]
We need to integrate the generated log file with Docker.
And also Its better not to log anything on Console on the deployment server!

So two more changes needs to be added

**Comment by maqboolkhan** [comment]
1) `IDA-QA` and `IDA-PROD` Docker envs (VM) will now store logs in `~/ida-qa-logs` and `~/ida-prod-logs` folders respectively.

2) While doing development locally, if one uses Docker he/she can also find logs in `~/ida-local-logs` folder and if one using `mvn` to run IDA locally then he/she can enable console logging by uncommenting https://github.com/dice-group/IDA/blob/8eccb8254b9c4de635b384d334278147a80bbad0/backend-server/src/main/resources/log4j2.xml#L15

Note: Console logging was switched off for performance reason as IDA writes logs on file anyway!

**Review by nikit91 — APPROVED** [review]
lgtm

**Pull request merged**